03-react/03-random-book-v2/src/main.tsx

```diff
@@ -0,0 +1,55 @@
+  
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+const appDiv = document.getElementById("app");
+
+if (!(appDiv instanceof HTMLDivElement)) {
+	throw new Error("No div with id 'app' found");
+}
+
+
+
+const App = () => {
+
+    const books: string[] = [
+	"Anna Karenina",
+	"To Kill a Mockingbird",
+	"The Great Gatsby",
+	"One Hundred Years of Solitude",
+	"A Passage to India",
+	"Invisible Man",
+	"Don Quixote",
+	"Beloved",
+	"Mrs. Dalloway",
+	"Things Fall Apart",
+	"Jane Eyre",
+	"The Color Purple",
+];
+
+const randomBook = books[Math.floor(Math.random() * books.length)]
+
+const colors:string[] = [
+    "red","green","blue","pink","gray","Aqua","BlueViolet","Crimson","DeepPink","Cyan"
+]
+
+const color = colors[Math.floor(Math.random() * colors.length)]
+const BC = colors[Math.floor(Math.random() * colors.length)]
+    
+	const [book,setBooks] = React.useState("");
+
+	return (
+		<main>
+           {//I want to change the body backgroundColor here but it is not working!
+           } 
+            <body style={{backgroundColor: BC}}>
+			<p style={{backgroundColor: color}}>{randomBook}</p>
+			<button onClick={() => setBooks(color+BC+book+1)}>
```

Comment:
That works because this will update the state, and therefore force your `App` component function to be called again, and you compute new random values there. But you actually do not use the `book` value itself,

To prepare for next exercises where we don't compute values randomly but get them from `<input>`s, and to learn the general pattern, I would like you to use two different states for the book title and the background color: `const [color, setColor] = useState(...)`, and `const [book, setBook] = useState(...)`.

Then, the computation of the random color/book should happen only when you set the state: `setBook(getRandomBook())`, not directly in the component function.